PROTO_tests/tests/random_test.py

```diff
@@ -246,6 +264,33 @@ def test_normal_hypothesis_testing(self):
         )
         assert good_fit_res.pvalue > 0.05
 
+    def test_poisson_hypothesis_testing(self):
+        # I tested this many times without a set seed, but with no seed
+        # it's expected to fail one out of every ~20 runs given a pval limit of 0.05.
+        rng = ak.random.default_rng(43)
+        num_samples = 10**4
+        lam = rng.uniform(0, 10)
+
+        sample = rng.poisson(lam=lam, size=num_samples)
+        count_dict = Counter(sample.to_list())
+
+        # the sum of exp freq must be within 1e-08, so use the cdf to find out how many
+        # elements we need to ensure we're within that tolerance
+        tol = 1e-09
+        num_elems = 5
+        while (1 - sp_stats.poisson.cdf(num_elems, mu=lam)) > tol:
+            num_elems += 5
+
+        obs_counts = np.array([0] * num_elems)
+        for k, v in count_dict.items():
+            obs_counts[k] = v
+
+        # use the probability mass function to get the probability of seeing each value
+        # and multiply by num_samples to get the expected counts
+        exp_counts = np.array([sp_stats.poisson.pmf(v, mu=lam) for v in range(num_elems)]) * num_samples
```

Comment:
This could also be written `sp_stats.poisson.pmf(range(num_elems), mu=lam) * num_samples`.